packages/cli/src/evaluation/test-runner/test-runner.service.ee.ts

```diff
@@ -78,19 +84,65 @@ export class TestRunnerService {
 		const executionId = await this.workflowRunner.run(data);
 		assert(executionId);
 
-		// Wait for the workflow to finish execution
+		// Wait for the execution to finish
 		const executePromise = this.activeExecutions.getPostExecutePromise(executionId);
 
 		return await executePromise;
 	}
 
+	/**
+	 * Run the evaluation workflow with the expected and actual run data.
+	 * @param evaluationWorkflow
+	 * @param expectedData
+	 * @param actualData
+	 * @private
+	 */
+	private async runTestCaseEvaluation(
+		evaluationWorkflow: WorkflowEntity,
+		expectedData: IRunData,
+		actualData: IRunData,
+	) {
+		// Prepare the evaluation wf input data.
+		// Provide both the expected data and the actual data
+		const evaluationInputData = {
+			json: {
+				originalExecution: expectedData,
+				newExecution: actualData,
+			},
+		};
+
+		// Prepare the data to run the evaluation workflow
+		const data = await getRunData(evaluationWorkflow, [evaluationInputData]);
+
+		// Trigger the evaluation workflow
+		const executionId = await this.workflowRunner.run(data);
+		assert(executionId);
+
+		// Wait for the execution to finish
+		const executePromise = this.activeExecutions.getPostExecutePromise(executionId);
+
+		return await executePromise;
+	}
+
+	private extractEvaluationResult(execution: IRun): IDataObject {
+		const lastNodeExecuted = execution.data.resultData.lastNodeExecuted;
+		assert(lastNodeExecuted, 'Could not find the last node executed in evaluation workflow');
+
+		// Extract the output of the last node executed in the evaluation workflow
+		// We use only the first main output
+		return execution.data.resultData.runData[lastNodeExecuted]?.[0]?.data?.main[0]?.[0]?.json ?? {};
```

Comment:
🥴

Reply:
Tried to make it a bit less confusing :)

Reply:
Yeah the issue is not this in this PR, it's the data structure itself